Mathlib/Probability/Kernel/RadonNikodym.lean

```diff
@@ -0,0 +1,425 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import Mathlib.Probability.Kernel.Disintegration.Density
+import Mathlib.Probability.Kernel.WithDensity
+
+/-!
+# Radon-Nikodym derivative and Lebesgue decomposition for kernels
+
+Let `γ` be a countably generated measurable space and `κ η : kernel α γ` be finite kernels.
+Then there exists a function `kernel.rnDeriv κ η : α → γ → ℝ≥0∞` jointly measurable on `α × γ`
+and a kernel `kernel.singularPart κ η : kernel α γ` such that
+* `κ = kernel.withDensity η (kernel.rnDeriv κ η) + kernel.singularPart κ η`,
+* for all `a : α`, `kernel.singularPart κ η a ⟂ₘ η a`,
+* for all `a : α`, `kernel.singularPart κ η a = 0 ↔ κ a ≪ η a`,
+* for all `a : α`, `kernel.withDensity η (kernel.rnDeriv κ η) a = 0 ↔ κ a ⟂ₘ η a`.
+
+Furthermore, the sets `{a | κ a ≪ η a}` and `{a | κ a ⟂ₘ η a}` are measurable.
+
+## Main definitions
+
+* `ProbabilityTheory.kernel.rnDeriv`: a function `α → γ → ℝ≥0∞` jointly measurable on `α × γ`
+* `ProbabilityTheory.kernel.singularPart`: a `kernel α γ`
+
+## Main statements
+
+* `ProbabilityTheory.kernel.mutuallySingular_singularPart`: for all `a : α`,
+  `kernel.singularPart κ η a ⟂ₘ η a`
+* `ProbabilityTheory.kernel.rnDeriv_add_singularPart`:
+  `kernel.withDensity η (kernel.rnDeriv κ η) + kernel.singularPart κ η = κ`
+* `ProbabilityTheory.kernel.measurableSet_absolutelyContinuous` : the set `{a | κ a ≪ η a}`
+  is Measurable
+* `ProbabilityTheory.kernel.measurableSet_mutuallySingular` : the set `{a | κ a ⟂ₘ η a}`
+  is Measurable
+
```

Comment:
Could you comment a little bit more on the implementation? (And in particular on the countably generated assumption, which is not relevant for the Radon-Nikodym theorem for measures but becomes necessary here unless I am mistaken).

Reply:
I added comments about the implementation and the assumption. I was not sure whether that precise countability assumption was necessary. The only reference I can find says it is, but without proof: in Probability and Stochastics, Cinlar states before theorem 4.44 that "the separability condition cannot be removed" (separability = our countably generated). There is no other comment on the question.